routes/drones.js

```diff
@@ -1,36 +1,81 @@
-const express = require('express');
+const express = require("express");
 const router = express.Router();
 
 // require the Drone model here
 
-router.get('/drones', (req, res, next) => {
+const Drone = require("../models/Drone.model");
+
+router.get("/drones", async (req, res, next) => {
   // Iteration #2: List the drones
   // ... your code here
+  try {
+    let drones = await Drone.find();
+    res.render("drones/list", { drones });
+  } catch (error) {
+    console.log(error);
+    next(error);
+  }
 });
 
-router.get('/drones/create', (req, res, next) => {
+router.get("/drones/create", (req, res, next) =>
   // Iteration #3: Add a new drone
   // ... your code here
-});
+  res.render("views/drones/create-form.hbs")
+);
 
-router.post('/drones/create', (req, res, next) => {
+router.post("/drones/create", async (req, res, next) => {
   // Iteration #3: Add a new drone
   // ... your code here
+  try {
+    const { name, propellers, maxSpeed } = req.body;
+    await Drone.create({ name, propellers, maxSpeed });
+    res.redirect("/drones");
+  } catch (error) {
+    console.log(error);
+    next(error);
+  }
 });
 
-router.get('/drones/:id/edit', (req, res, next) => {
+router.get("/drones/:id/edit", async (req, res, next) => {
   // Iteration #4: Update the drone
   // ... your code here
+  try {
+    const { id } = req.params;
+    const book = await Drone.findById(id);
+    res.render("views/drones/update-form.hbs", drone);
```

Comment:
The same thing here, you passed the wrong path to your view and that's why you're getting an error on that pages.